hikari/impl/rate_limits.py

```diff
@@ -438,25 +438,41 @@ class ExponentialBackOff:
     Parameters
     ----------
     base : builtins.float
-        The base to use. Defaults to `2`.
+        The base to use. Defaults to `2.0`.
     maximum : typing.Optional[builtins.float]
         If not `builtins.None`, then this is the max value the backoff can be
         in a single iteration before an `asyncio.TimeoutError` is raised.
-        Defaults to `64` seconds.
+        Defaults to `64.0` seconds.
     jitter_multiplier : builtins.float
-        The multiplier for the random jitter. Defaults to `1`.
+        The multiplier for the random jitter. Defaults to `1.0`.
         Set to `0` to disable jitter.
     initial_increment : builtins.int
         The initial increment to start at. Defaults to `0`.
+
+    Raises
+    ------
+    ValueError
+        If an `builtins.int` that's too big to be represented as a
+        `builtins.float` is passed in place of a field that's annotated as
+        `builtins.float` or if `initial_increment` is invalid for the provided
+        back-off rules (e.g. too large).
     """
 
-    __slots__: typing.Sequence[str] = ("base", "increment", "maximum", "jitter_multiplier")
+    __slots__: typing.Sequence[str] = ("_is_frozen", "base", "increment", "maximum", "jitter_multiplier")
```

Comment:
When do we hit a floating point issue?

Can we use `decimal.Decimal` here instead?

Reply:
decimal.Decimal also has maximum size limits on certain impls and this won't necessarily be bigger than the maximum floating point number limit